lib/kafka_ex.ex

```diff
@@ -144,14 +144,10 @@ defmodule KafkaEx do
     max_bytes     = Keyword.get(opts, :max_bytes, @max_bytes)
     auto_commit   = Keyword.get(opts, :auto_commit, true)
 
-    offset = case offset do
-      nil -> last_offset = offset_fetch(worker_name, %KafkaEx.Protocol.OffsetFetch.Request{topic: topic, partition: partition}) |> KafkaEx.Protocol.OffsetFetch.Response.last_offset
-        if last_offset <= 0 do
-          0
-        else
-          last_offset
-        end
-      _   -> offset
+    unless offset do
+      # retrieve last stored offset from kafka
+      last_offset = offset_fetch(worker_name, %KafkaEx.Protocol.OffsetFetch.Request{topic: topic, partition: partition}) |> KafkaEx.Protocol.OffsetFetch.Response.last_offset
+      offset = max(last_offset + 1, 0)
```

Comment:
`last_offset` can return 0 if it a new `consumer_group` and none of the consumers in the group has consumed anything.